src/modules/CurrencyList/Currency/Currency.tsx

```diff
@@ -0,0 +1,42 @@
+import { FC } from "react";
+import { CurrencyType } from "../../../types";
+import { Wrapper } from "../../../components/Wrapper";
+import styles from "./currency.module.css";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+type CurrencyProps = {
+  currency: CurrencyType;
+  setPresentCurrency: React.Dispatch<React.SetStateAction<null | CurrencyType>>;

```

Comment:
Nie jest to błąd ale warto rozważyć aby nie przekazywać funkcji setState, tylko funkcję typu `(currency: CurrencyType) => void`

Reply:
State setter owinięty w funkcję. Funkcja przekazana przez props.